packages/bot/src/stores/toolbar-store.js

```diff
@@ -25,9 +26,11 @@ export default class ToolbarStore {
 
     @action.bound
     onToolboxToggle() {
-        // eslint-disable-next-line no-underscore-dangle
-        const toolbox = Blockly.derivWorkspace.toolbox_;
+        const toolbox        = Blockly.derivWorkspace.toolbox_; // eslint-disable-line
+        const toolbox_width  = toolbox.HtmlDiv.clientWidth;
         this.is_toolbox_open = !this.is_toolbox_open;
+
+        scrollWorkspace(Blockly.derivWorkspace, toolbox_width, true, !this.is_toolbox_open);
```

Comment:
The workspace will be scrolled even when blocks were not in the position of the most left. Is there anyway we can determine whether those blocks were at the most left so that can we will only scroll when those blocks cover the position of toolbox.

Reply:
@aaron-binary @sam-binary dont need to make it complicated i belive.